lib/xapi-stdext-unix/unixext.ml

```diff
@@ -161,12 +161,17 @@ let string_of_file file_path = Buffer.contents (buffer_of_file file_path)
 let atomic_write_to_file fname perms f =
   let module Filenameext = Xapi_stdext_std.Filenameext in
   let tmp = Filenameext.temp_file_in_dir fname in
-  Unix.chmod tmp perms;
+  let tmp_perms = 0o600 in
```

Comment:
* I believe `tmp` has already permissions `0600` 
* https://github.com/xapi-project/stdext/blob/master/lib/xapi-stdext-std/filenameext.ml#L22
* `fd` is opened with the existing permissions (0600) because the file exists
* Is it possible that only changing permissions to `perm` was missing?

Reply:
The call fails when passing a `perm` that's read-only. It seems the ocaml documentation (and comment) doesn't match the implementation.